templates/GenericTemplate.hs

```diff
@@ -239,7 +239,7 @@ happyMonad2Reduce k nt fn j tk st sts stk =
          let drop_stk = happyDropStk k stk
 #if defined(HAPPY_ARRAY)
              off = indexShortOffAddr happyGotoOffsets st1
-             off_i = PLUS(off,nt)
+             off_i = if LT(off, (unbox_int happyMinOffset)) then PLUS(off, PLUS(nt, ILIT(65536))) else PLUS(off, nt)
```

Comment:
It would be really nice if we could only omit this conditional if we actually need it.  Happy could figure out whether the grammar requires overflow and `#define` something that would get checked here.

Not essential, but I'm always a bit paranoid about adding some extra overhead into the central part of the algorithm like this.

Reply:
I agree. I'll work on that when I have some time (probably this coming weekend).